core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java

```diff
@@ -534,15 +498,16 @@ public int size() {
          return (int) size;
       }
 
-      @Override
-      public boolean isEmpty() {
-         boolean empty = cacheSet.isEmpty();
-         // Check the store if the data container was empty
-         if (empty) {
-            empty = persistenceManager.size() == 0;
-         }
-         return empty;
-      }
+      // TODO: this can't be added in until stores size method is guaranteed to not count expired entries
```

Comment:
Sure sounds good. I had thought about doing that with iterator, but didn't think of `publishKeys`. I think it has just been too long since i looked at this.